core/src/main/scala/kafka/raft/KafkaNetworkChannel.scala

```diff
@@ -124,8 +124,18 @@ class KafkaNetworkChannel(
     }
 
     def onComplete(clientResponse: ClientResponse): Unit = {
-      val response = if (clientResponse.authenticationException != null) {
-        errorResponse(request.data, Errors.CLUSTER_AUTHORIZATION_FAILED)
+      val response = if (clientResponse.versionMismatch != null) {
+        error(s"Request $request failed due to unsupported version error",
+          clientResponse.authenticationException)
```

Comment:
It should pass `clientResponse.versionMismatch` rather than `clientResponse.authenticationException`